package/kernel/mac80211/files/lib/netifd/wireless/mac80211.sh

```diff
@@ -646,7 +646,7 @@ find_phy() {
 		}
 	}
 	[ -n "$macaddr" ] && {
-		for phy in $(ls /sys/class/ieee80211 2>/dev/null); do
+		for phy in /sys/class/ieee80211/*; do
```

Comment:
You should add
```
phy=${phy##*/}
```
to strip the path and only keep the phy name.
And you have to also cover the case of `/sys/class/ieee80211/` being empty or not even existing. In that case the loop content will be called once with phy="/sys/class/ieee80211/*"

Reply:
so replace with find?